tests/compile_time_tests.cpp

```diff
@@ -53,4 +57,38 @@ static_assert
     !std::integral< float >
 );
 
+static_assert
+(
+    std::contiguous_iterator< std::initializer_list< int >::iterator >
+);
+
+static_assert
+(
+    std::input_iterator< std::initializer_list< int >::iterator >
+);
+
+struct Triplet
+{
+    int a;
+    int b;
+    int c;
+
+    [[ nodiscard ]] constexpr int sum() const noexcept { return a + b + c; }
+};
+
+// testing std::construct_at
+static_assert
+(
+    []()
+    {
+        union Union {
+            Triplet container;
+            char    c;
+        };
+        Union u;
```

Comment:
This default-constructs the `container` anyway, so the line below is a no-op.

Please add a custom constructor to `Triplet` and default constructor to `Union`.